### PR TITLE
Deliver actual charge links data to BFF - instead of mocked data

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Initially code from @BjarkeMeier's team demo of CQRS. Eventually it has transitioned into:
- [x]  Delivering real charge links to BFF from new query stack (instead of mocked data)
- [x] Refactor sorting to align with business requirements:
* Order by `Charge Type`
* Then order by `Charge Id`, ascending
* And then by `StartDate` descending
- [x] CIM value naming applied for ChargeType, so e.g. "D01" is used instead of "1" to indicate a subscription.

### Notes: 
* Adding Web API integration tests will be done in a subsequent PR.
* Wiring Charges Web API and application insights will also be done in a subsequent PR.

### Out-of-scope:
- Name of the charge owner (it is hardcoded as it is currently not present in the domain)

## References

* #628
* #827

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202112161505 Remove redundant ChargePoint to Charge FK.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- Drop FK with dynamic name on column ChargeOperationRowId
+-- No easy way, so remove both constraints and then recreate the one that should stay (but this time with a name!)
+DECLARE @dropCommand nvarchar(1000)
+SELECT @dropCommand = 'alter table ' + schema_name(Schema_id)+'.'+ object_name(parent_object_id) + '  DROP CONSTRAINT  ' +  name
+from sys.foreign_keys
+where schema_name(Schema_id)+'.'+ object_name(parent_object_id) = 'Charges.ChargePoint'
+
+EXEC sp_executesql @dropCommand
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202112161505 Remove redundant ChargePoint to Charge FK.sql:8]: I see the same command as part of an earlier merged sql [file](https://github.com/Energinet-DataHub/geh-charges/blob/ee4bd1855047e2c04985ff69208c0fc06abcd2bb/source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202112081220%20Remove%20unsued%20ChargePoint%20columns.sql#L13). Is there a reason to do this again? 🤔

Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202112161505 Remove redundant ChargePoint to Charge FK.sql]: Concluded this script to be a duplicate. Removed it in 822f2388

**Comment by BjarkeMeier** [comment]
As ChargeType=1 is an internal choice it shouldn't be exposed. We should publish the CIM "D01" value.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs [hunk 2]
```diff
@@ -12,25 +12,25 @@
...
 {
-    [UnitTest]
-    public class ChargeTypeTests
+    public class Data : IData
     {
-        /// <summary>
-        /// See https://github.com/Energinet-DataHub/geh-charges/issues/933 regarding naming decisions.
-        /// </summary>
-        [Fact]
-        public void Types_HaveCorrectNames()
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs:29]: Why is this test being removed?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Model/ChargeLink/ChargeLinkDtoTests.cs [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
-namespace GreenEnergyHub.Charges.Tests.WebApi.Model.ChargeLink
-{
-    [UnitTest]
-    public class ChargeLinkDtoTests
-    {
-        /// <summary>
-        /// See https://github.com/Energinet-DataHub/geh-charges/issues/933 regarding naming decisions.
-        /// </summary>
-        [Fact]
-        public void Props_HaveCorrectNames()
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Model/ChargeLink/ChargeLinkDtoTests.cs:29]: Why is this test being removed?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Controllers/ChargeLinksControllerTests.cs [hunk 4]
```diff
@@ -1,57 +0,0 @@
...
-using FluentAssertions;
-using GreenEnergyHub.Charges.WebApi.Controllers;
-using Microsoft.AspNetCore.Mvc;
-using Xunit;
-using Xunit.Categories;
-
-namespace GreenEnergyHub.Charges.Tests.WebApi.Controllers
-{
-    [UnitTest]
-    public class ChargeLinksControllerTests
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Controllers/ChargeLinksControllerTests.cs:25]: I assume that removed controller tests are replaced - or will be - by other tests(?)

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/EnvironmentSettingNames.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    /// <summary>
+    /// Contains names of settings used by the web api.
+    /// </summary>
+    public static class EnvironmentSettingNames
+    {
+        // Function
+        public const string AzureWebJobsStorage = "AzureWebJobsStorage";
+
+        // Environment specific settings
+        public const string AppInsightsInstrumentationKey = "APPINSIGHTS_INSTRUMENTATIONKEY";
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/EnvironmentSettingNames.cs:26]: Did you add this to the function host configuration in terraform?

**Review by BjarkeMeier — COMMENTED** [review]
LGTM but left some comments and questions.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/EnvironmentSettingNames.cs (hunk 5 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.WebApi/EnvironmentSettingNames.cs:26]: Wiring up with app insights will be done in later.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs (hunk 2 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs:29]: I am not sure. I think you did the change before christmas? ☎️ ?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Controllers/ChargeLinksControllerTests.cs (hunk 4 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Controllers/ChargeLinksControllerTests.cs:25]: They will be replaced.
The integration tests for the ChargeLinksController will follow this PR
And the unit tests for the controller will follow after an upgrade of the Web API project to EFCore 6.0

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs (hunk 2 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.QueryApi/Data.cs]: added the test again with 34ddd142

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Model/ChargeLink/ChargeLinkDtoTests.cs (hunk 3 above)
Comment (prtandrup) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/WebApi/Model/ChargeLink/ChargeLinkDtoTests.cs]: Re-added with https://github.com/Energinet-DataHub/geh-charges/pull/768/commits/34ddd1423297f9fcf02ba5fcf8270826f5323b91. Nice spotted.

**Comment by prtandrup** [comment]
> As ChargeType=1 is an internal choice it shouldn't be exposed. We should publish the CIM "D01" value. (I'm referring to the PR description)

Fixed with https://github.com/Energinet-DataHub/geh-charges/pull/768/commits/bb95fc39119d1b44adcf59dfa5d983728c1e747a CIM value naming is now in place.

**Review by Renetnielsen — APPROVED** [review]

**Review by prtandrup — APPROVED** [review]
Work/PR initiated by @BjarkeMeier, finished by me, @prtandrup. Bjarke has reviewed it, but cannot approve (as he is the PR creator), hence I approve it.

**Pull request merged**